src/panoptoindexconnector/implementations/microsoft_graph_implementation.py

```diff
@@ -331,6 +384,41 @@ def set_principals_to_user(config, panopto_content, target_content):
             target_content["acl"].append(acl)
 
 
+def set_principals_to_user_group(config, panopto_content, target_content):
+    """
+    Set session permission to user group
+    """
+
+    target_content["acl"] = []
+
+    for ec in get_unique_user_group_external_contexts(config, panopto_content):
+
+        panopto_user_group_identifier = ec.get("ExternalId")
+        aad_group_id = None
+
+        # Try to get user group id from user_groups list
+        if panopto_user_group_identifier in user_groups:
+            aad_group_id = user_groups.get(panopto_user_group_identifier)
+        # If user group doesn't exist in list, try to get from AAD calling API
+        else:
+            # Get user group from AAD
+            aad_user_group_info = get_aad_user_group_info(config, panopto_user_group_identifier)
+
+            if aad_user_group_info:
+                aad_group_id = aad_user_group_info["id"]
```

Comment:
We are sure this is valid because "Id" is always set , correct?

Reply:
That is true. Id is group identifier and it should be always set.